src/Reflection/RepositoryFindMethodsClassReflectionExtension.php

```diff
@@ -23,30 +23,34 @@ public function setBroker(Broker $broker): void
 
 	public function hasMethod(ClassReflection $classReflection, string $methodName): bool
 	{
-		if (
-			!$classReflection->hasNativeMethod($methodName)
-			&& in_array(Repository::class, $classReflection->getParentClassesNames(), true)
-		) {
-			if (strpos($methodName, 'findOneBy') === 0) {
-				$propertyName = lcfirst(substr($methodName, 9));
-			} elseif (strpos($methodName, 'findBy') === 0) {
-				$propertyName = lcfirst(substr($methodName, 6));
-			} else {
-				return false;
-			}
-
-			// ensure that a property with that name exists on the model, as there might
-			//  be methods starting with find[One]By... with custom implementations on
-			//  inherited repositories
-			$className = $classReflection->getName();
-			$modelName = ClassNamingUtility::translateRepositoryNameToModelName($className);
-
-			$modelReflection = $this->broker->getClass($modelName);
-			if ($modelReflection->hasProperty($propertyName)) {
-				return true;
-			}
+		if (!in_array(Repository::class, $classReflection->getParentClassesNames(), true)) {
+			return false;
 		}
-		return false;
+
+		if ($classReflection->hasNativeMethod($methodName)) {
+			return false;
+		}
+
+		if (strpos($methodName, 'findOneBy') === 0) {
```

Comment:
Could we also add the countBy magic method here? Would also solve #28

Reply:
done.